src/shared/modules/auth/helpers.js

```diff
@@ -0,0 +1,133 @@
+import { isObject } from 'lodash'
+import { AUTH_LOGGING_PREFIX, AUTH_STORAGE_LOGS } from './constants'
+import { isAuthLoggingEnabled, isAuthDebuggingEnabled } from './settings'
+import { saveAs } from 'file-saver'
+
+export const authLog = (msg, type = 'log') => {
+  if (!isAuthLoggingEnabled) return
+  if (!['log', 'error', 'warn'].includes(type)) return
+  const log = `${AUTH_LOGGING_PREFIX} [${new Date().toISOString()}] ${msg}`
+  const logs = sessionStorage.getItem(AUTH_STORAGE_LOGS) || ''
+  const logsLines = logs.split('\n')
```

Comment:
this could be problematic if there are line breaks in the individual logged messages (seems like it would only affect the limit, not the functionality)

Reply:
Good catch. Removed newlines in log messages to make them easier to read and to get around this issue